functests/sctp/sctp.go

```diff
@@ -483,3 +494,67 @@ func setupEgress(namespace, fromPod, toPod string, port int32) error {
 	_, err := client.Client.NetworkPolicies(namespace).Create(policy)
 	return err
 }
+
+func findSCTPNodeSelector() (string, error) {
+	mcList, err := client.Client.MachineConfigs().List(metav1.ListOptions{})
+	Expect(err).ToNot(HaveOccurred())
+
+	for _, mc := range mcList.Items {
+		if enablesSCTP(mc) {
+			newSCTPNodeSelector, err := findSCTPNodeSelectorByMCLabel(mc.ObjectMeta.Labels["machineconfiguration.openshift.io/role"])
+			if err != nil {
+				return "", err
+			}
+			return newSCTPNodeSelector, nil
+		}
+	}
+
+	return "", errors.New("cannot find a machine configuration that enables SCTP")
+}
+
+func enablesSCTP(mc mcfgv1.MachineConfig) bool {
+	firstPathFound := false
+	secondPathFound := false
+
+	for _, file := range mc.Spec.Config.Storage.Files {
+		if file.Path == sctpBlacklistPath {
+			firstPathFound = true
+		}
+		if file.Path == sctpLoadPath {
+			secondPathFound = true
+		}
+	}
+
+	return firstPathFound && secondPathFound
+}
+
+func findSCTPNodeSelectorByMCLabel(mcLabel string) (string, error) {
+	mcpList, err := client.Client.MachineConfigPools().List(metav1.ListOptions{})
+	Expect(err).ToNot(HaveOccurred())
+
+	for _, mcp := range mcpList.Items {
+		for _, lsr := range mcp.Spec.MachineConfigSelector.MatchExpressions {
+			for _, value := range lsr.Values {
+				if value == mcLabel {
+					for key, label := range mcp.Spec.NodeSelector.MatchLabels {
+						newSCTPNodeSelector := key + "=" + label
+
+						// check if there are at least two nodes or skip
+						sctpNodeList, err := client.Client.Nodes().List(metav1.ListOptions{
```

Comment:
No need to do this (and the check of the len) here anymore.
This function declares it's purpouse in the name but here we are doing more than that. Also, one node is enough.
Just return the selector and leave to the caller the responsability to use it.

Reply:
should i check the number of nodes on the caller side?
and is 1 node enough?